### PR TITLE
add bg and flipping

### PR DESCRIPTION
**Pull request by jmerkow**
image

image

### CODE REVIEW

**Comment by jmerkow** [comment]
@ironchefpython is it possible to rotate/flip a bp without displaying it in the canvas?

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "license": "GPLV3",
   "dependencies": {
     "@azure/storage-blob": "^12.14.0",
+    "browserify": "^17.0.0",
```
Comment (ironchefpython) [inline package.json]: What is this for?

File: bp-image.html [hunk 2]
```diff
@@ -144,48 +183,133 @@
...
+    import { Blueprint } from "./blueprint.js";
     import { styles, TRANSFORMATIONS } from "./constants.js";
     const canvas = document.getElementById('myCanvas');
 
     const queryParams = new URLSearchParams(window.location.search);
-    const bps = queryParams.get('bp');
-    
-    // Size of HTML document (same as pageHeight/pageWidth in screenshot).
+    var bps = queryParams.get('bp');
+    const rotate = queryParams.get('rotate', false) ?? false;
```
Comment (ironchefpython) [inline bp-image.html:268]: Where is rotate counter-clockwise and horizontal flip?

File: blueprint-view.js [hunk 3]
```diff
@@ -139,37 +139,79 @@ class BlueprintView extends Blueprint {
...
+        const scaledHeight = cropHeight * scale;
+        const offsetX = Math.floor((this.canvas.width - scaledWidth) / 2);
+        const offsetY = Math.floor((this.canvas.height - scaledHeight) / 2);
+        this.ctx.globalAlpha = 0.7; // Set opacity to 90%
+        this.ctx.drawImage(img, x, y, cropWidth, cropHeight, offsetX, offsetY, scaledWidth, scaledHeight);
+        this.ctx.globalAlpha = 1; // Reset opacity
+
+
+        this.xoffset = 0;
+        this.yoffset = 0;
```
Comment (ironchefpython) [inline blueprint-view.js]: These assignments do nothing?

File: package.json (hunk 1 above)
Comment (jmerkow) [reply package.json]: Yeah need to remove it

File: bp-image.html (hunk 2 above)
Comment (jmerkow) [reply bp-image.html:268]: I figure they can just rotate multiple times to get what they want

File: blueprint-view.js (hunk 3 above)
Comment (jmerkow) [reply blueprint-view.js]: Ya

**Pull request merged**